index.bs

```diff
@@ -1715,3 +1705,220 @@ When [=a browsing context is discarded=], run the following steps:
  1. [=Emit an event=] with |body| and |parent id|.
 
 </div>
+
+## Script ## {#module-script}
+
+### Definition ### {#module-script-definition}
+
+[=remote end definition=] and [=local end definition=]
+```
+Realm = text;
+
+RealmOrBrowsingContext = (
+  {realm: Realm} //
+  {context: BrowsingContext}
+)
+
+RealmInfo = {
+  realm: Realm,
+  type: RealmType,
+  origin: text,
+}
+
+RealmType = (
+  "window" / "worker" / "sharedworker" / "serviceworker" / "worklet" / text
+)
+```
+
+Each [=realm=] has an associated <dfn export>realm id</dfn>, which is a string
+uniquely identifying that realm. This is implicitly set when the realm is
+created.
+
+<div algorithm>
+To <dfn>get a realm</dfn> given |realm id|:
+
+ 1. If |realm id| is null, return [=success=] with data null.
+
+ 1. If there is no realm with [=realm id=] |realm id| return
+    [=error=] with [=error code=] [=no such frame=]
+
+ 1. Let |realm| be the browsing context with id |realm id|.
+
+ 1. Return [=success=] with data |realm|
+
+Issue: This has the wrong error code
+
+</div>
+
+<div algorithm>
+To <dfn>get the realm info</dfn> given |environment settings|:
+
+ 1. Let |realm| be |environment settings|' [=realm execution context=]'s Realm component.
+
+ 1. Let |realm id| be the [=realm id=] for |realm|.
+
+ 1. Run the steps under the first matching condition:
+
+   <dl>
+     <dt>The [=Realm/global object=] specified by |environment settings| is a [=Window=] object
+     <dd>
+      1. If [=Realm/global object=] specified by |environment settings| has a [=Document=]
```

Comment:
Why this early return? When will this scenario occur? If it happens very rarely, then it's likely clients will come to assume the return value is a string, and just break in rare conditions.

If there are ways of using a realm that don't make sense when the document isn't yet fully active, maybe those commands should fail in some way instead? But if it's rare, the problem will remain.

Reply:
I removed this condition for now.